packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -464,6 +464,18 @@ export async function runNpmInstall(
       commandArgs = args
         .filter(a => a !== '--prefer-offline')
         .concat(['install', '--no-audit', '--unsafe-perm']);
+      if (
+        nodeVersion?.major === 16 &&
```

Comment:
Does this need to check the `npm` version? Do we even need to do an `if` at all? I just tested `npm@6.14.2` with `--legacy-peer-deps` and it succeeded without error or any warnings. Was this `npm` issue just with 8.6.0? The `npm` version that ships with Node 18 won't be an issue?

Reply:
@cb1kenobi We don't have the npm version here unfortunately. The issue is with 8.6.0 or newer.

You can read more details in slack: https://vercel.slack.com/archives/C041Z5U9C5B

Also see this document: https://www.notion.so/npm-install-exited-with-1-54368abd9798467f8296dfed3cc33258

Reply:
The problem is the Node.js version doesn't necessarily correspond to the npm version.

Getting the npm version is pretty quick and easy:

```javascript
const [ major, minor ] = execFileSync('npm', ['--version'], { encoding: 'utf-8' }).trim().split('.').map(n => ~~n);
if (major > 8 || (major === 8 && minor >= 6)) {
  commandArgs.push('--legacy-peer-deps');
}
```

It's relatively cheap compared to the `npm install` that will happen next in the code. I measure about 185ms on my machine.

Or, just always set `--legacy-peer-deps` as it doesn't seem to cause issue with npm 6. :)

Reply:
> Or, just always set --legacy-peer-deps as it doesn't seem to cause issue with npm 6

I think that is the best solution however I'm using node 16 as a way to prevent patching this when we release node 18 as my comment mentions.

Since users have to opt into node 18 (when the time comes) we will no longer need this patch